internal/arp/arp.go

```diff
@@ -0,0 +1,153 @@
+package arp
+
+import (
+	"bytes"
+	"errors"
+	"net"
+	"sync"
+
+	"github.com/golang/glog"
+	"github.com/mdlayher/arp"
+	"github.com/mdlayher/ethernet"
+)
+
+// Announce is used to "announce" new IPs mapped to the node's MAC address.
+type Announce struct {
+	hardwareAddr net.HardwareAddr
+	client       *arp.Client
+	c            chan *arp.Packet
+	ips          map[string]net.IP // map containing IPs we should announce
+	sync.RWMutex                   // protects ips
+}
+
+// New return an initialized Announce.
+func New(ip net.IP) (*Announce, error) {
+	ifi, err := interfaceByIP(ip)
+	if err != nil {
+		return nil, err
+	}
+	client, err := arp.Dial(ifi)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Announce{
+		hardwareAddr: ifi.HardwareAddr,
+		client:       client,
+		c:            make(chan *arp.Packet),
+		ips:          make(map[string]net.IP),
+	}, nil
+}
+
+// Start starts the announcer, making it listen on the interface for ARP requests.
+func (a *Announce) Start() {
+	// Read packet from the wire.
+	go func() {
```

Comment:
Curious: why the goroutines here?

Reply:
Grand plan that fell apart in the last minute. There is no reason why this should be very different from: https://raw.githubusercontent.com/mdlayher/arp/master/cmd/proxyarpd/main.go
i.e. just a `for` loop reading and responding.